ladybug/datacollectionimmutable.py

```diff
@@ -0,0 +1,203 @@
+# coding=utf-8
+"""Immutable versions of the Ladybug Data Collections.
+
+Note that all of the methods or properties on an immutable collection that
+return another data collection will return a collection that is mutable.
+
+The only exceptions to this rule are:
+duplicate() - which will always return an exact copy of the collection including
+    its mutabiliy.
+get_aligned_collection() - which follows the mutability of the starting collection
+    by default but includes an parameter to override this.
+to_immutable() - which clearly always returns an immutable version of the collection
+
+Note that the to_mutable() method on the immutable collections can always be used to
+get a mutable version of an immutable collection.
+"""
+from __future__ import division
+
+from .datacollection import HourlyDiscontinuousCollection, HourlyContinuousCollection, \
+    DailyCollection, MonthlyCollection, MonthlyPerHourCollection
+
+from collections import Iterable
+
+
+class HourlyDiscontinuousCollectionImmutable(HourlyDiscontinuousCollection):
+    """Immutable Discontinous Data Collection at hourly or sub-hourly intervals."""
+    _mutable = False
+
+    @property
+    def values(self):
+        """The Data Collection's list of numerical values."""
+        return tuple(self._values)
+
+    @values.setter
+    def values(self, values):
+        try:
+            self._values
```

Comment:
a cleaner way is probably trying `hasattr(self, '_values')`

Reply:
you can probably simplify this as:

```python
if hasattr(self, '_values'):
   raise AttributeError(self._mutable_message())

self._check_values(values)
self._values = tuple(values)

```

Reply:
same comments for all the classes in this module.

Reply:
Ahhhh!  That's much nicer.  Ok, let me do that.

Reply:
Man, that is so much better.  I was so used to using `hasattr` as an incorrect placeholder for `isinstance` that I forgot it has a real useful purpose!